ipv8/messaging/anonymization/tunnel.py

```diff
@@ -72,9 +72,9 @@ def could_be_udp_tracker(data):
     @staticmethod
     def could_be_dht(data):
         try:
-            if len(data) > 1 and data[0:1] == 'd' and data[-1:] == 'e':
```

Comment:
Oh, you found a bug. `data` is a bytes object, so it should be compared to `b'd'` and `b'e'` (something we missed when porting to Python 3).

Other than that, the old code was correct (slicing over a `bytes` produces `bytes`, indices produce `int`):

```python
a = b"data"
print(a[0] == b"d")
>> False
print(a[0:1] == b"d")
>> True
```

Reply:
Ohhhh, ok. Gotcha

Reply:
Done